packages/react/src/components/popover/useOnClickOutside.spec.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  beforeAll,
+  beforeEach,
+  describe,
+  expect,
+  it,
+  jest,
+} from '@jest/globals';
+import { Mock } from 'jest-mock';
+import { useEffect } from 'react';
+import { useOnClickOutside } from './useOnClickOutside';
+
+jest.mock('react', () => ({
+  useEffect: jest.fn((fn: () => unknown) => fn()),
+  useRef: (current: unknown) => ({ current }),
+}));
+
+describe('useOnClickOutside', () => {
+  const addEventListener = jest.fn();
+  const removeEventListener = jest.fn();
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  beforeAll(() => {
+    Object.defineProperties(document, {
+      addEventListener: { value: addEventListener },
+      removeEventListener: { value: removeEventListener },
+    });
+  });
+
+  it('should add event listener to document', () => {
+    useOnClickOutside(() => {}, [{ current: {} as Element }]);
+
+    expect(addEventListener).toBeCalledTimes(1);
+  });
+
+  it('should remove event listener from document on effect cleanup', () => {
+    let cleanup = (): unknown => void 0;
+    (useEffect as Mock<any>).mockImplementationOnce(
+      (fn: () => () => unknown) => (cleanup = fn())
+    );
+
+    useOnClickOutside(() => {}, [{ current: {} as Element }]);
+    cleanup();
+
+    expect(removeEventListener).toBeCalledTimes(1);
+  });
+
+  it('should call onClickOutside on outside click', () => {
+    const onClickOutside = jest.fn();
+    const target = { current: { contains: () => false } as any };
+    const event = { target };
+    addEventListener.mockImplementationOnce((_, fn: (_: unknown) => void) =>
```

Comment:
The underscore convention denotes unused args?

Reply:
Yup, avoids `eslint` complaining too. Any name starting with `_` it will ignore.